java/client/src/org/openqa/selenium/By.java

```diff
@@ -138,6 +139,27 @@ public WebElement findElement(SearchContext context) {
    */
   public abstract List<WebElement> findElements(SearchContext context);
 
+  /**
+   * Nest a child By selector into this one, allowing greater composability.
+   *
+   * @param child The child selector to nest inside the current one (searching for
+   *              elements inside the result of this search).
+   * @return
+   */
+  public By nest(By child) {
+    final By parent = this;
+    return new By() {
+      public List<WebElement> findElements(SearchContext context) {
+        List<WebElement> results = new ArrayList<>();
```

Comment:
Maybe a `LinkedList` would be better? I'm not sure.

Reply:
No, `LinkedList` is almost never worth using.